tests/ext/drv/test_ext.c

```diff
@@ -0,0 +1,489 @@
+
+    if (program_info_client == NULL) {
+        status = STATUS_NO_MEMORY;
+        goto Exit;
+    }
+
+    program_info_client->nmr_binding_handle = nmr_binding_handle;
+    program_info_client->client_module_id = client_registration_instance->ModuleId->Guid;
+
+Exit:
+    if (NT_SUCCESS(status)) {
+        *provider_binding_context = program_info_client;
+        program_info_client = NULL;
+    } else
+        ebpf_free(program_info_client);
+    return status;
+}
+
+/**
+ * @brief Callback invoked when a Program Information NPI client detaches.
+ *
+ * @param client_binding_context Provider module's context for binding with the client.
+ * @retval STATUS_SUCCESS The operation succeeded.
+ * @retval STATUS_INVALID_PARAMETER One or more parameters are invalid.
+ */
+NTSTATUS
+_test_ebpf_extension_program_info_provider_detach_client(void* provider_binding_context)
+{
+    NTSTATUS status = STATUS_SUCCESS;
+
+    ebpf_free(provider_binding_context);
+
+    return status;
+}
+
+void
+test_ebpf_extension_program_info_provider_unregister()
+{
+    test_ebpf_extension_program_info_provider_t* provider_context = &_test_ebpf_extension_program_info_provider_context;
+    NTSTATUS status = NmrDeregisterProvider(provider_context->nmr_provider_handle);
+    if (status == STATUS_PENDING)
+        NmrWaitForProviderDeregisterComplete(provider_context->nmr_provider_handle);
+}
+
+NTSTATUS
+test_ebpf_extension_program_info_provider_register()
+{
+    test_ebpf_extension_program_info_provider_t* local_provider_context;
+    ebpf_extension_data_t* extension_data;
+    ebpf_program_data_t* program_data;
+
+    NTSTATUS status = STATUS_SUCCESS;
+
+    extension_data = (ebpf_extension_data_t*)_test_ebpf_extension_program_info_provider_characteristics
+                         .ProviderRegistrationInstance.NpiSpecificCharacteristics;
+    program_data = (ebpf_program_data_t*)extension_data->data;
+    program_data->program_info->program_type_descriptor.program_type = EBPF_PROGRAM_TYPE_TEST;
+
+    local_provider_context = &_test_ebpf_extension_program_info_provider_context;
+
+    status = NmrRegisterProvider(
+        &_test_ebpf_extension_program_info_provider_characteristics,
+        local_provider_context,
+        &local_provider_context->nmr_provider_handle);
+    if (!NT_SUCCESS(status))
+        goto Exit;
+
+Exit:
+    if (!NT_SUCCESS(status))
+        test_ebpf_extension_program_info_provider_unregister();
+
+    return status;
+}
+
+//
+// Hook Provider.
+//
+
+/**
+ * @brief Callback invoked when a eBPF hook NPI client attaches.
+ *
+ * @param[in] nmr_binding_handle NMR binding between the client module and the provider module.
+ * @param[in] provider_context Provider module's context.
+ * @param[in] ClientRegistrationInstance Client module's registration data.
+ * @param[in] client_binding_context Client module's context for binding with provider.
+ * @param[in] client_dispatch_table Client module's dispatch table. Contains the function pointer
```

Comment:
The argument is currently named `client_dispatch`.  Make doxygen and the actual arg be consistent.
Also add doxygen lines for provider_binding_context and provider_dispatch